ceto/ais_adapter.py

```diff
@@ -0,0 +1,557 @@
+        dim_c (float): Dim c [m]
+        dim_d (float): Dim d [m]
+        speed (float): Current speed [kn]
+        draft (float): Current draft [kn]
+        latitude (float): Current position (latitude) [deg]
+        longitude (float): Current position (longitude) [deg]
+
+    Raises:
+        ValueError: If validation of inputted parameters fail,
+        which they do if they are not deemed resonable for further guesstimations.
+
+    Returns:
+        dict: Vessel data
+    """
+    # Validation
+    if not _validate_dims(dim_a, dim_b, dim_c, dim_d):
+        raise ValueError(
+            "Dims (dim_a, dim_b, dim_c, dim_d) are not deemed reasonable. No estimations can be made."
+        )
+
+    # Ship type
+    ceto_ship_type = _map_type_of_ship_and_cargo_type_to_ceto_ship_type(
+        type_of_ship_and_cargo_type
+    )
+
+    # Main dimensions
+    length, beam = dim_a + dim_b, dim_c + dim_d
+    design_draft = _guesstimate_design_draft(draft, beam)
+    design_speed = _guesstimate_design_speed(length, ceto_ship_type, speed)
+
+    # Vessel size (DWT)
+    vessel_size = _guesstimate_vessel_size_as_DeadWeightTonnage(
+        ceto_ship_type, length, beam, draft
+    )
+
+    # Engine parameters
+    number_of_engines = _guesstimate_number_of_engines(ceto_ship_type)
+    engine_power = _guesstimate_engine_MCR(ceto_ship_type, vessel_size, design_speed)
+    engine_type = _guesstimate_engine_type(ceto_ship_type)
+    engine_fuel_type = _guesstimate_engine_fuel_type(
+        ceto_ship_type, latitude, longitude
+    )
+
+    # Vessel size as GT
+    if ceto_ship_type in (
+        "ferry-pax",
+        "ferry-ropax",
+        "cruise",
+        "yacht",
+        "miscellaneous-fishing",
+        "service-tug",
+        "offshore",
+        "service-other",
+        "miscellaneous-other",
+    ):
+        vessel_size = _guesstimate_vessel_size_as_GrossTonnage(
+            length, beam, design_draft
+        )
+    elif ceto_ship_type == "liquified_gas_tanker":
+        vessel_size = _guesstimate_vessel_size_as_CubicMetres(
+            length, beam, design_draft
+        )
+
+    # Assemble output dictionary
+    vessel_data = {
+        # Fixed values that will not be inferred from AIS data but are reasonable static assumptions
+        **{
+            "double_ended": False,
+            "propulsion_engine_age": "after_2000",
+        },
+        # Guesstimated values
+        **{
+            "type": ceto_ship_type,
+            "length": length,
+            "beam": beam,
+            "size": vessel_size,
+            "design_draft": design_draft,
+            "design_speed": design_speed,
+            "number_of_propulsion_engines": number_of_engines,
+            "propulsion_engine_power": engine_power,
+            "propulsion_engine_type": engine_type,
+            "propulsion_engine_fuel_type": engine_fuel_type,
+        },
+    }
+
+    return vessel_data
+
+
+EARTH_RADIUS = 6367.0 * 1000.0
+
+
+def _rhumbline(
```

Comment:
Hmm, not a good one. It does calculate the rhumbline distance (loxodrome) rather than the great circle distance. (see for example the section about "Rhumb lines" [here](http://www.movable-type.co.uk/scripts/latlong.html)) Since I expect this function to be used for subsequent AIS message (or similar) there really is no need to calculate the great circle distance?

Reply:
Added this source in the docstring of the function.